### PR TITLE
Build prover with byte arrays of params

### PR DESCRIPTION
**Pull request by adityapk00**

### CODE REVIEW

File: zcash_proofs/src/prover.rs
Comment (nuttycom) [inline zcash_proofs/src/prover.rs:69]: Could you please return `Result` or `Option` rather than panicing? The caller can panic, but this should be a pure function.

**Review by nuttycom — CHANGES_REQUESTED** [review]

File: zcash_proofs/src/prover.rs
Comment (nuttycom) [reply zcash_proofs/src/prover.rs:69]: @adityapk00 if you're able to resolve this in the next day or two, we'll be able to get this PR into the 4.1.0 release.

Comment (adityapk00) [reply zcash_proofs/src/prover.rs]: Thanks, @nuttycom - I refactored it a bit, so it uses the same `load_parameters` method, which also checks the hashes.

**Review by nuttycom — APPROVED** [review]
I think it's better to just make parse_parameters public and document that it may panic here; I pushed such a change back to your branch.

**Comment by adityapk00** [comment]
Thanks @nuttycom . Can you point me to the change you made, please? I think I may have force-pushed over the change while I was fighting `rustfmt`

File: zcash_proofs/src/prover.rs
Comment (nuttycom) [inline zcash_proofs/src/prover.rs]:
```suggestion
    pub fn from_bytes(spend_param_bytes: &[u8], output_param_bytes: &[u8]) -> Self {
```

**Review by nuttycom — APPROVED** [review]

**Comment by nuttycom** [comment]
@adityapk00 there you go - @str4d requested a minor name change to `from_bytes` which I also included here.

**Comment by adityapk00** [comment]
Thanks! I updated the doc tests as well with the new name. 

Do I need to squash all the commits before merging?

**Comment by nuttycom** [comment]
Sure, if you'd like to clean up the commit history that'd be great.

**Comment by adityapk00** [comment]
Great! Done. Cleaned up the branch, and I think its ready for merge. 

I can't see a merge button, so I assume someone else has to merge it? 
(Sorry for all the questions, I can never seem to remember github processes)

**Comment by nuttycom** [comment]
Yup, I will merge once CI passes.

**Comment by adityapk00** [comment]
Thanks! On an unrelated note, librustzcash isn't compiling for me using the latest rust nightly compiler. I assume that's also related to the "nightly lint" faillures in the CI checks. Is this a known issue?

**Comment by nuttycom** [comment]
Yes, there's a known issue with the nightlies right now.

**Pull request merged**
